mavros/src/plugins/waypoint.cpp

```diff
@@ -513,10 +509,16 @@ class WaypointPlugin : public plugin::PluginBase {
 	void connection_cb(bool connected) override
 	{
 		lock_guard lock(mutex);
-		if (connected)
+		if (connected) {
 			shedule_pull(BOOTUP_TIME_DT);
-		else
+
+			if (!wp_nh.getParam("enable_partial_push",enable_partial_push)) {
+				enable_partial_push = m_uas->is_ardupilotmega();
+			}
```

Comment:
I just checked NodeHandle, and better to use `wp_nh.hasParam()`: http://docs.ros.org/jade/api/roscpp/html/classros_1_1NodeHandle.html#aadd6a7c41de08f54a913e97f26a3af5e